### ISSUE TITLE
Recursive clone not possible du to permission denied

### ISSUE THREAD

**Issue by backupandforkacc** [Kaljurand/K6nele-service#4]
I thought the keys are only important when I upload something to github, but I am getting this error when trying to clone as described:

git clone --recursive git@github.com:Kaljurand/K6nele-service.git
Cloning into 'K6nele-service'...
git@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

**Comment by backupandforkacc** [comment]
Just for the next one to know:

1. Add github account to Android studio:
https://stackoverflow.com/questions/68659576/how-to-add-a-github-account-in-android-studio-artic-fox

2. Add public key to github account (under settings)

**Comment by Kaljurand** [comment]
Yes, it's a known issue to which I don't have a good solution yet. The issue being that convenient read-only cloning on the command line requires authentication via a GitHub account.

The problem is that ``.gitmodules`` specifies the speechutils-URL using the ``git@github`` notation, which in turn requires authentication.

One work-around (which is maybe not acceptable in all use cases) is:

```
git clone git://github.com/Kaljurand/K6nele-service.git
cd K6nele-service/
git clone git://github.com/Kaljurand/speechutils.git
cd speechutils/
git checkout e1ce5d5 # this must point to the currently used changeset of speechutils
```
